poe-tasks/gradle-connector-tasks.toml

```diff
@@ -80,6 +80,57 @@ help = "Get the language of the connector from its metadata.yaml file. Use with
 cmd = """yq eval '.data.connectorBuildOptions.baseImage' $PWD/metadata.yaml"""
 help = "Get the base image of the connector from its metadata.yaml file."
 
+[tasks.get-local-version]
+cmd = """yq eval '.data.dockerImageTag' $PWD/metadata.yaml"""
+help = "Get the version of the connector from its metadata.yaml file. Use with -qq to get just the version."
+
+[tasks.create-connector-registry-artifacts]
+shell = '''
+set -eu
+
+# Define required and optional files
+REQUIRED_FILES=(
+  "metadata.yaml"
+  "icon.svg"
+)
+
+OPTIONAL_FILES=(
+  "build.gradle"
+  "build.gradle.kts"
+)
+
+VERSION=$(poe -qq get-local-version)
+CONNECTOR_NAME=$(poe -qq get-connector-name)
+
+# Create build directory
+mkdir -p "build/releases/${VERSION}"
+
+# Clear existing contents
+rm -rf "build/releases/${VERSION}"/*
+
+# Copy required files
+for file in "${REQUIRED_FILES[@]}"; do
+  cp "$file" "build/releases/${VERSION}/"
+done
+
+# Copy optional files if they exist
+for file in "${OPTIONAL_FILES[@]}"; do
+  if [ -f "$file" ]; then
+    cp "$file" "build/releases/${VERSION}/"
+  fi
+done
+
+# Copy distribution tar files if they exist (from build/distributions)
```

Comment:
For Java connectors, don't assume the build might or might not exists. Instead, first nuke the build/distributions directory and then run `poe gradle distTar` to create the .tar file. Fail if there aren't any matching files after running distTar, and otherwise copy it and add a .sha256 file.